glusterd2/commands/volumes/volume-smartvol-txn.go

```diff
@@ -48,15 +48,35 @@ func txnPrepareBricks(c transaction.TxnCtx) error {
 			}
 
 			// LV Creation
-			err = deviceutils.CreateLV(b.VgName, b.TpName, b.LvName, b.Size)
-			if err != nil {
-				c.Logger().WithError(err).WithFields(log.Fields{
-					"vg-name": b.VgName,
-					"tp-name": b.TpName,
-					"lv-name": b.LvName,
-					"size":    b.Size,
-				}).Error("lvcreate failed")
-				return err
+			if b.Type == "arbiter" {
+				// All the calculations should be in MB
+				arbiterSize := (4 / 1024) * (b.Size / req.AverageFileSize)
```

Comment:
one more question, what happens if the `req.AverageFileSize` is 0 from rest call?

Reply:
Nice catch. Have fixed this case.